### PR TITLE
feat: Support --mount option on Mac

### PR DESCRIPTION
**Pull request by cfxegbert**
New integration tests do not run correctly on OS X Ventura so could not complete testing.  We need to fix the integration tests in another ticket.  See https://matrix.to/#/!oTWUEgcTDkFeeuGXEo:gitter.im/$sHUfOebw04q5cobvWx8T5jodBeBfAnv4CihS4cn5K_I?via=gitter.im&via=matrix.org&via=nixos.dev

### CODE REVIEW

**Comment by daviessm** [comment]
Nice one! I don't suppose there's a crate we could use to get the mounts so we don't have to worry so much about `unsafe` is there?

File: src/fs/mounts/macos.rs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+use std::os::raw::{c_char, c_int};
+use std::os::unix::ffi::OsStrExt;
+use std::path::PathBuf;
+use libc::{__error, getfsstat, MNT_NOWAIT, statfs};
+use crate::fs::mounts::{Error, MountedFs};
+
+/// Get a list of all mounted filesystem
+pub fn mounts() -> Result<Vec<MountedFs>, Error> {
+    // Passing a null pointer and zero bufsize will return the number of mounts
+    let mut count: i32 = unsafe { getfsstat(ptr::null_mut(), 0, MNT_NOWAIT) };
```
Comment (daviessm) [inline src/fs/mounts/macos.rs:12]: If the `unsafe` is unavoidable please can you add SAFETY comments as per https://rust-lang.github.io/rust-clippy/master/#/undocumented_unsafe_blocks?

File: src/fs/mounts/macos.rs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    let mut mntbuf = Vec::<statfs>::new();
+    if count > 0 {
+        mntbuf.resize_with(count as usize, || unsafe { mem::zeroed() });
```
Comment (daviessm) [inline src/fs/mounts/macos.rs:15]: Same - please document `unsafe` block

File: src/fs/mounts/macos.rs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        let bufsize = mntbuf.len() * mem::size_of::<statfs>();
+        count = unsafe { getfsstat(mntbuf.as_mut_ptr(), bufsize as c_int, MNT_NOWAIT) };
```
Comment (daviessm) [inline src/fs/mounts/macos.rs:17]: Same - please document `unsafe` block

File: src/fs/mounts/macos.rs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        // Resize if the mount table has shrunk since last call
+        if count >= 0 {
+            mntbuf.truncate(count as usize);
+        }
+    }
+    if count < 0 {
+        return Err(Error::GetFSStatError(unsafe { *__error() }));
```
Comment (daviessm) [inline src/fs/mounts/macos.rs:24]: Same - please document `unsafe` block

File: src/fs/mounts/macos.rs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    let mut mounts = Vec::with_capacity(count as usize);
+    for mnt in &mntbuf {
+        let mount_point = OsStr::from_bytes(
+            unsafe { CStr::from_ptr(mnt.f_mntonname.as_ptr().cast::<c_char>()) }.to_bytes()
```
Comment (daviessm) [inline src/fs/mounts/macos.rs:30]: Same - please document `unsafe` block

File: src/fs/mounts/macos.rs [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        );
+        let dest = PathBuf::from(mount_point);
+        let fstype = unsafe { CStr::from_ptr(mnt.f_fstypename.as_ptr().cast::<c_char>()) }
```
Comment (daviessm) [inline src/fs/mounts/macos.rs:33]: Same - please document `unsafe` block

File: src/fs/mounts/macos.rs [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+            .to_string_lossy()
+            .into();
+        let source = unsafe { CStr::from_ptr(mnt.f_mntfromname.as_ptr().cast::<c_char>()) }
```
Comment (daviessm) [inline src/fs/mounts/macos.rs:36]: Same - please document `unsafe` block

**Review by daviessm — CHANGES_REQUESTED** [review]
Just a few comments on documentation of `unsafe` blocks. The change seems logical overall.

**Review by cafkafk — APPROVED** [review]
Nice, thanks for refactoring the mess I made of `lib.rs`. Cool to see we get to leverage `1.70` already with `OnceLock`. Hopefully we get clap merged very soon so the comment can be addressed.

**Pull request merged**